app/v1/controllers/AmendLossClaimController.scala

```diff
@@ -82,4 +92,38 @@ class AmendLossClaimController @Inject()(val authService: EnrolmentsAuthService,
       case DownstreamError => InternalServerError(Json.toJson(errorWrapper))
     }
   }
+
+
+  private def createAuditDetails(nino: String,
+                                 claimId: String,
+                                 statusCode: Int,
+                                 request: JsValue,
+                                 correlationId: String,
+                                 userDetails: UserDetails,
+                                 errorWrapper: Option[ErrorWrapper] = None,
+                                 response: Option[JsValue]
+                                ) = {
+    val auditResponse =
+      errorWrapper
+        .map { wrapper =>
+          AuditResponse(statusCode, Some(wrapper.auditErrors), None)
+        }
+        .getOrElse(AuditResponse(statusCode, None, response))
+
+    AmendLossClaimAuditDetail(
+      userType = userDetails.userType,
+      agentReferenceNumber = userDetails.agentReferenceNumber,
+      nino = nino,
+      claimId = claimId,
+      request = request,
+      correlationId,
+      auditResponse)
+  }
+
+  private def auditSubmission(details: AmendLossClaimAuditDetail)
```

Comment:
Could this be made generic for all endpoints?

Reply:
The only reason it isn't is the auditType and transactionName are different for all of them, and I would argue that e.g. introducing abstract methods to supply these for each controller might reduce clarity given how simple the method is anyway.